site/docs/v1/tech/tutorials/integrate-laravel-api.md

```diff
@@ -0,0 +1,245 @@
+
+After it is done, change into the `fusionauth-example-laravel-api` directory and install [`tymon/jwt-auth`](https://github.com/tymondesigns/jwt-auth), a library that adds JWT capabilities to Laravel.
+
+```shell
+cd fusionauth-example-laravel-api
+./vendor/bin/sail composer require tymon/jwt-auth
+```
+
+Add some authentication routes to `routes/api.php`. 
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/49a4e7c3e3610403c6a00246bf5b997bc44b5bd4.diff %}
+```
+
+Create `app/Http/Controllers/AuthController.php` to hold the authentication process.
+
+```php
+{% remote_include https://raw.githubusercontent.com/FusionAuth/fusionauth-example-laravel-api/main/laravel/app/Http/Controllers/AuthController.php %}
+```
+
+Laravel uses something called [Guards](https://laravel.com/docs/10.x/authentication#adding-custom-guards) to protect your endpoints, so we need to tell it about the new Guard provided from that library.
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/5fb7134a70e1adee39f2d1d64ce4ed5ac7303a47.diff %}
+```
+
+To make the library available for use, publish its configuration by running the command below.
+
+```shell
+./vendor/bin/sail artisan vendor:publish --provider="Tymon\JWTAuth\Providers\LaravelServiceProvider"
+```
+
+### Editing Files
+
+The Laravel installer already brings some useful resources for many applications. We'll have to edit a few of them and add some more.
+
+First, remove the need for users to have a password by editing `app/Models/User.php`.
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/e4298a10437ff396fe3dfc02c3bcf85f310921a0.diff %}
+```
+
+Change the created migration at `database/migrations/2014_10_12_000000_create_users_table.php`.
+
+```diff
+{% remote_include https://github.com/FusionAuth/fusionauth-example-laravel-api/commit/9c670115fa991d2219120a88ccaaf94b9e5feab1.diff %}
+```
+
+You may remove all other migrations, as you won't need them.
+
+```shell
+rm database/migrations/2014_10_12_100000_create_password_reset_tokens_table.php
+rm database/migrations/2019_08_19_000000_create_failed_jobs_table.php
+rm database/migrations/2019_12_14_000001_create_personal_access_tokens_table.php
+```
+
+Run the remaining migrations to create the necessary tables in your database.
+
+```shell
+./vendor/bin/sail artisan migrate
+```
+
+### Adding Files
+
+By default, Laravel only allows JWTs that correspond to existing users in your database, but one of the greatest benefits of using FusionAuth is to have a single source of truth of user management. So, you would want your API to automatically provision new users when it receives a trusted JWT from FusionAuth. Even though [`tymon/jwt-auth`](https://github.com/tymondesigns/jwt-auth) is a great library, it doesn't provide an easy way of doing these kinds of customizations, so you'll have to create some classes to do them.
```

Comment:
Hmmm. Why do we need to provision users at all? Can't you deliver resources (like a simple message json, as the ruby on rails API does) without users?

Reply:
Unfortunately not (at least not an easy way) as there are a lot of calls to retrieve the user from the request, so this is why I had to do all of this.